### ISSUE TITLE
[java] ExhaustiveSwitchHasDefault issue with final fields not initialized in constructor

### ISSUE THREAD

**Issue by kohlschuetter** [pmd/pmd#5670]
**Affects PMD Version:**  7.12.0

**Rule:** ExhaustiveSwitchHasDefault
https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#exhaustiveswitchhasdefault

**Description:**
ExhaustiveSwitchHasDefault cannot be applied because the compiler complains that a final field was not initialized (for the impossible, missing default case). 

Moreover, in combination with CheckStyle, which has a "MissingSwitchDefault" rule, this new PMD rule is impossible to apply.

**Code Sample demonstrating the issue:**
```java
import java.io.PrintStream;

public class Test {
  private final PrintStream out;

  public enum ProcessStream {
    STDOUT,
    STDERR
  }

  public Test(ProcessStream stream) {
    switch (stream) {
      case STDOUT:
        this.out = System.out;
        break;
      case STDERR:
        this.out = System.err;
        break;
      default:
        // Without this "default" section, Eclipse complains with:
        // The blank final field out may not have been initialized.
        // Note that a problem regarding missing 'default:' on 'switch' has been suppressed,
        // which is perhaps related to this problem
        throw new IllegalStateException();
    }
  }
}
```

**Expected outcome:**

Since the assumption that the compiler is doing its job right is wrong, I think we should at least work around this limitation, or disable the rule by default.

**Running PMD through:** Maven
